packages/react/slot/src/Slot.tsx

```diff
@@ -13,28 +13,22 @@ const Slot = React.forwardRef<HTMLElement, SlotProps>((props, forwardedRef) => {
   const { children, ...slotProps } = props;
   const childLength = React.Children.count(children);
 
-  if (childLength === 1) {
-    return (
-      <SlotClone {...slotProps} ref={forwardedRef}>
-        {children}
-      </SlotClone>
-    );
-  }
-
-  return (
-    <>
-      {React.Children.map(children, (child) => {
-        if (React.isValidElement(child) && child.type === Slottable) {
-          return (
-            <SlotClone {...slotProps} ref={forwardedRef}>
-              {child.props.children}
-            </SlotClone>
-          );
-        }
-
-        return child;
-      })}
-    </>
+  return childLength === 1 ? (
+    <SlotClone {...slotProps} ref={forwardedRef}>
+      {children}
+    </SlotClone>
+  ) : React.Children.toArray(children).some(isSlottable) ? (
+    React.Children.map(children, (child) => {
+      return isSlottable(child) ? (
+        <SlotClone {...slotProps} ref={forwardedRef}>
+          {child.props.children}
+        </SlotClone>
+      ) : (
+        child
+      );
+    })
+  ) : (
+    React.Children.only(children)
```

Comment:
This is now saying that if the length is greater than `1` and none of the children contain a `Slottable` then it is the consumer code providing more than one child so we throw the `React.Children.only` error.